mcnode/net.go

```diff
@@ -322,6 +327,85 @@ func (node *Node) netConns() []p2p_pstore.PeerInfo {
 	return peers
 }
 
+func (node *Node) netPing(ctx context.Context, pid p2p_peer.ID) (dt time.Duration, err error) {
+	err = node.doConnectPeer(ctx, pid)
+	if err != nil {
+		return
+	}
+
+	ch, err := node.ping.Ping(ctx, pid)
+	if err != nil {
+		return
+	}
+
+	select {
+	case <-ctx.Done():
+		err = ctx.Err()
+		return
+
+	case dt = <-ch:
+		return
+	}
+}
+
+func (node *Node) netIdentify(ctx context.Context, pid p2p_peer.ID) (nid NetIdentify, err error) {
+	if node.status == StatusOffline {
+		return nid, NodeOffline
+	}
+
+	if node.host.Network().Connectedness(pid) == p2p_net.Connected {
+		goto identify
```

Comment:
There is no misguided fetish against goto in golang, and that's why the language explicitly supports it.
Besides, it's the right way to do it here.